clang_format.py

```diff
@@ -116,29 +166,60 @@ def clang_format_path() -> Path:
     return clang_format_file
 
 
-def main():
+def main(argv: Optional[Sequence[str]] = None) -> Union[int, None]:
+    parser = argparse.ArgumentParser(description="Arguments for pre commit.")
+    parser.add_argument(
+        "version",
+        choices=(
+            "3.5.0",
```

Comment:
read these choices from CLANG_FORMAT_SHAS ?

Reply:
updated.

Reply:
A new function is created to get the shas from the list and the function is used in choice param for version arg.